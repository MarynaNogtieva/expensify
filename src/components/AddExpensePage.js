import React from 'react';

const AddExpensePage = () => {
  return(
    <div>
    this is from my AddExpensePage  component
   </div>
  )
};

export default AddExpensePage;