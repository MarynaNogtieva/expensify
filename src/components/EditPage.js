import React from 'react';

const EditPage = () => {
  return(
    <div>
    this is from my EditPage component
   </div>
  )
};

export default EditPage;