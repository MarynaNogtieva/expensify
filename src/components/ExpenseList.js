import React from 'react';
import { connect } from 'react-redux'

const ExpenseList = ({expenses, filters}) => {
  return (
    <div>
      <h1>Expense List</h1>
      {
        expenses.map((expense) => {
          return <p key={ expense.id }>{ expense.description }</p>
        })
      }
      { filters.text }
    </div>
  )
};

const mapStateToProps = (state) => {
  return {
    expenses: state.expenses,
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseList);
