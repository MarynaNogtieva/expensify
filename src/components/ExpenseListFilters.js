import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates'
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate   } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };

  //destructure from react-dates library
  onDatesChange = ({startDate, endDate}) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({calendarFocused}));
  }
  
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  }

  onSortChange = (e) => {
    if(e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  }
  render() {
    console.log(this.props.filters)
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
            type='text'
            value={ this.props.filters.text }
            onChange={this.onTextChange}
            className="text-input"
            placeholder="Search expenses"
            />
          </div>
          <div>
            <select
            className="select"
            value={this.props.filters.sortBy}
            onChange={this.onSortChange}>
              <option value='date'>Date</option>
              <option value='amount'>Amount</option>
            </select>
          </div>
          <div>
            <DateRangePicker
            startDate={this.props.filters.startDate}
            endDate={this.props.filters.endDate}
            onDatesChange={this.onDatesChange}
            focusedInput={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            showClearDates={true}
            numberOfMonths={1}
            isOutsideRange={()=> false} //allow go back in time
            />
          </div>
        </div>
        
        
        
      </div>
    )
  }
}


const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const matDispatchToProps = (dispatch) => {
  return {
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dospatch(setEndDate(endDate))
  }
}

export default connect(mapStateToProps, matDispatchToProps)(ExpenseListFilters);