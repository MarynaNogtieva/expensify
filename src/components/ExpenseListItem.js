import React from 'react';


const ExpenseListItem = ({ expenses }) => {
  return (
    <div>
      {
        expenses.map((expense) => {
          return (
          <p key={ expense.id }>
            { expense.description } &nbsp;
            { expense.amount } &nbsp;
            { expense.createdAt }
          </p>
          )
        })
      }
    </div>
  );
};

export default ExpenseListItem;