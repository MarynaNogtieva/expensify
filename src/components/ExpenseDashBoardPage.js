import React from 'react';
import ExpenseList from './ExpenseList';

const ExpenseDashBoardPage = () => {
  return(
   <div>
    <ExpenseList />
   </div>
  );
 };

 export default ExpenseDashBoardPage;