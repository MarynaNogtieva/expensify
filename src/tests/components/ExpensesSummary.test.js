import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import numeral from 'numeral';


test('should correctly rener EspensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={23454} />)
  expect(wrapper).toMatchSnapshot();
});

test('should correctly rener EspensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={23454} />)
  expect(wrapper).toMatchSnapshot();
});