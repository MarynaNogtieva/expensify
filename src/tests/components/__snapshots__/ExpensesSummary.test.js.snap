// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly rener EspensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $234.54
  </h1>
</div>
`;

exports[`should correctly rener EspensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $234.54
  </h1>
</div>
`;
